packer/communicator/azureVmCustomScriptExtension/communicator.go

```diff
@@ -110,85 +121,77 @@ func (c *comm) sleepSec(d time.Duration) {
-	reqManager := c.config.AzureServiceRequestManager
+func (c *comm) requestCustomScriptExtension() (*vm.ResourceExtension, error) {
 
 	log.Println("Requesting resource extensions...")
-	requestData := reqManager.ListResourceExtensions()
-	resp, err := reqManager.Execute(requestData)
-
-	if err != nil {
-		return nil, err
-	}
-
-	list, err := response.ParseResourceExtensionList(resp.Body)
 
+	list, err := vm.NewClient(c.config.ManagementClient).GetResourceExtensions()
 	if err != nil {
 		return nil, err
 	}
 
 	log.Println("Searching for CustomScriptExtension...")
-	ext := list.FirstOrNull("CustomScriptExtension")
-	log.Printf("CustomScriptExtension: %v\n\n", ext)
-
-	if ext == nil {
-		err = fmt.Errorf("CustomScriptExtension is nil")
-		return nil, err
+	for _, ext := range list {
+		if ext.Name == "CustomScriptExtension" {
+			log.Printf("CustomScriptExtension: %v\n\n", ext)
+			return &ext, nil
+		}
 	}
 
-	return ext, nil
+	return nil, fmt.Errorf("Couldn't find CustomScriptExtension, am I too old or is Azure broken?")
 }
 
 func (c *comm) updateRoleResourceExtension(
 	nameOfReference, nameOfPublisher, nameOfExtension, versionOfExtension, state string,
-	params []azureservice.ResourceExtensionParameterValue) error {
+	params []vm.ResourceExtensionParameter) error {
 
-	reqManager := c.config.AzureServiceRequestManager
+	client := c.config.ManagementClient
 
 	serviceName := c.config.ServiceName
 	vmName := c.config.VmName
 
 	log.Println("Updating Role Resource Extension...")
 
-	requestData := reqManager.UpdateRoleResourceExtensionReference(serviceName, vmName, nameOfReference, nameOfPublisher, nameOfExtension, versionOfExtension, state, params)
-	err := reqManager.ExecuteSync(requestData)
+	role := vm.Role{}
+	vmutils.AddAzureVMExtensionConfiguration(&role,
+		nameOfExtension, nameOfPublisher, versionOfExtension, nameOfReference, state, []byte{}, []byte{})
+	// HACK-paulmey: clean up later
+	(*role.ResourceExtensionReferences)[0].ParameterValues = params
 
-	if err != nil {
+	if err := retry.ExecuteAsyncOperation(client, func() (management.OperationId, error) {
+		return vm.NewClient(client).UpdateRole(serviceName, vmName, vmName, role)
+	}); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (c *comm) buildParams(runScript string) (params []azureservice.ResourceExtensionParameterValue) {
-	storageAccountName, storageAccountKey := c.config.StorageServiceDriver.GetProps()
-
-	account := "{\"storageAccountName\":\"" + storageAccountName + "\",\"storageAccountKey\": \"" + storageAccountKey + "\"}"
+func (c *comm) buildParams(runScript string) []vm.ResourceExtensionParameter {
 
 	scriptfile := "{\"fileUris\": [" + c.uris + "], \"commandToExecute\":\"powershell -ExecutionPolicy Unrestricted -file " + runScript + "\"}"
+	account := "{\"storageAccountName\":\"" + c.config.StorageAccountName + "\",\"storageAccountKey\": \"" + c.config.StorageAccountKey + "\"}"
```

Comment:
just sayin but why can't you just create an anonymous struct here and marshal it to json?